### PR TITLE
Gcore cdn deploy

### PR DESCRIPTION
**Pull request by temoffey**

### CODE REVIEW

**Comment by Neilpang** [comment]
please resolve the conflicts.

**Comment by temoffey** [comment]
complete

File: deploy/gcore_cdn.sh [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+  _cca="$4"
+  _cfullchain="$5"
+
+  _debug _cdomain "$_cdomain"
+  _debug _ckey "$_ckey"
+  _debug _ccert "$_ccert"
+  _debug _cca "$_cca"
+  _debug _cfullchain "$_cfullchain"
+
+  _fullchain=$(awk 1 ORS='\\n' "$_cfullchain")
```
Comment (Neilpang) [inline deploy/gcore_cdn.sh]: can you replace awk with sed, cut or grep etc.
do not use awk.

File: deploy/gcore_cdn.sh [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+    if [ -z "$Le_Deploy_gcore_cdn_password" ]; then
+      _err "Please define the target password: export DEPLOY_GCORE_CDN_PASSWORD=password"
+      return 1
+    fi
+  else
+    Le_Deploy_gcore_cdn_password="$DEPLOY_GCORE_CDN_PASSWORD"
+    _savedomainconf Le_Deploy_gcore_cdn_password "$Le_Deploy_gcore_cdn_password"
+  fi
+
+  if ! [ -x "$(command -v jq)" ]; then
```
Comment (Neilpang) [inline deploy/gcore_cdn.sh]: do not use jq

File: deploy/gcore_cdn.sh [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+    _err "Please install the package jq: sudo apt-get install jq"
+    return 1
+  fi
+
+  _info "Get authorization token"
+  _request="{ \"username\": \"$Le_Deploy_gcore_cdn_username\", \"password\": \"$Le_Deploy_gcore_cdn_password\" }"
+  _debug _request "$_request"
+  _response=$(curl -s -X POST https://api.gcdn.co/auth/signin -H "Content-Type:application/json" -d "$_request")
```
Comment (Neilpang) [inline deploy/gcore_cdn.sh]: do not use curl
please use `_post()` function instead.

File: deploy/gcore_cdn.sh [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+  if [ "$_sslDataNew" != "$_sslDataAdd" ]; then
+    _err "Error CDN resource update"
+    return 1
+  fi
+
+  if [ -z "$_sslDataOld" ] || [ "$_sslDataOld" = "null" ]; then
+    _info "Not found old SSL certificate"
+  else
+    _info "Delete old SSL certificate"
+    _response=$(curl -s -X DELETE "https://api.gcdn.co/sslData/$_sslDataOld" -H "Authorization:Token $_token")
```
Comment (Neilpang) [inline deploy/gcore_cdn.sh]: use `_post()` function to send `DELETE` request.

**Comment by bkfon** [comment]
complete

**Comment by temoffey** [comment]
complete

File: deploy/gcore_cdn.sh [hunk 5]
```diff
@@ -0,0 +1,140 @@
...
+  fi
+
+  _info "Find CDN resource with cname $_cdomain"
+  export _H2="Authorization:Token $_token"
+  _response=$(_get "https://api.gcdn.co/resources")
+  _debug _response "$_response"
+  _regex=".*(\"id\".*?\"cname\":\"$_cdomain\".*?})"
+  _regex="\"cname\":\"$_cdomain\""
+  _debug _regex "$_regex"
+  _resource=$(echo "$_response" | sed 's/},{/},\n{/g' | grep -E "$_regex")
```
Comment (Neilpang) [inline deploy/gcore_cdn.sh]: do not use `grep -E`.   please use `_egrep_o()` instead.

**Pull request merged**